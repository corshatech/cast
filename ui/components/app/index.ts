export * from "./Layout/Layout";
export * from "./FeatureComponent/FeatureComponent";
