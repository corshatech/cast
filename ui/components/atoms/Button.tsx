export type ButtonProps = JSX.IntrinsicElements["button"] & {
  variant?: string;
  size?: string;
};

export const Button = ({ children, ...args }: ButtonProps) => {
  return <button {...args}>{children}</button>;
};
