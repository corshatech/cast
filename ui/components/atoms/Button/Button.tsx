import clsx from 'clsx';

const classes = {
  size: (val: ButtonProps['size']) =>
    val === 'sm'
      ? 'px-10 py-2'
      : val === 'md'
      ? 'px-[70px] py-3'
      : 'px-[75px] py-3.5',
  variant: (val: ButtonProps['variant']) =>
    val === 'primary'
      ? 'bg-corsha-brand-blue text-white border-2 border-corsha-brand-blue hover:bg-corsha-brand-mid-blue hover:border-[#104457] focus:bg-corsha-brand-blue focus:border-2 focus:border-corsha-brand-green disabled:bg-[#A1A9B4] disabled:border-[#A1A9B4] disabled:cursor-not-allowed'
      : val === 'secondary'
      ? 'bg-white text-primary border-2 border-corsha-brand-blue hover:bg-highlight-blue focus:border-2 focus:border-corsha-brand-green disabled:border-[#A1A9B4] disabled:text-[#A1A9B4] disabled:hover:bg-white disabled:cursor-not-allowed'
      : 'bg-transparent',
};

export type ButtonProps = JSX.IntrinsicElements['button'] & {
  variant?: 'primary' | 'secondary';
  size?: 'sm' | 'md' | 'lg';
};

export const Button = ({ children, variant, size, ...props }: ButtonProps) => {
  return (
    <button
      className={clsx(
        'rounded-md space-x-3 leading-none inline-block',
        classes.variant(variant),
        classes.size(size),
      )}
      {...props}
    >
      {children}
    </button>
  );
};
