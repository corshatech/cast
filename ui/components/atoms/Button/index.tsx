/* Copyright 2023 Corsha.
   Licensed under the Apache License, Version 2.0 (the 'License');
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at
   http://www.apache.org/licenses/LICENSE-2.0
   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an 'AS IS' BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License. */

import clsx from 'clsx';

const classes = {
  size: (val: ButtonProps['size']) =>
    val === 'sm'
      ? 'px-10 py-2'
      : val === 'md'
      ? 'px-[70px] py-3'
      : 'px-[75px] py-3.5',
  variant: (val: ButtonProps['variant']) =>
    val === 'primary'
      ? 'bg-corsha-brand-blue text-white border-2 border-corsha-brand-blue hover:bg-corsha-brand-mid-blue hover:border-[#104457] focus:bg-corsha-brand-blue focus:border-2 focus:border-corsha-brand-green disabled:bg-[#A1A9B4] disabled:border-[#A1A9B4] disabled:cursor-not-allowed'
      : val === 'secondary'
      ? 'bg-white text-primary border-2 border-corsha-brand-blue hover:bg-highlight-blue focus:border-2 focus:border-corsha-brand-green disabled:border-[#A1A9B4] disabled:text-[#A1A9B4] disabled:hover:bg-white disabled:cursor-not-allowed'
      : 'bg-transparent',
};

export type ButtonProps = JSX.IntrinsicElements['button'] & {
  variant?: 'primary' | 'secondary';
  size?: 'sm' | 'md' | 'lg';
};

export const Button = ({ children, variant, size, ...props }: ButtonProps) => {
  return (
    <button
      className={clsx(
        'rounded-md space-x-3 leading-none inline-block',
        classes.variant(variant),
        classes.size(size),
      )}
      {...props}
    >
      {children}
    </button>
  );
};
