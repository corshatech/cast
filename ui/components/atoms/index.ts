export * from './Button/Button';
export * from './Typography/Typography';
