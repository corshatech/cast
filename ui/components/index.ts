export * from './atoms';
export * from './app';
