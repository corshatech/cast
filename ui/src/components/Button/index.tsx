import React from "react";
import css from "./Button.module.scss";

interface ButtonProps {
  /**
   * Is this the principal call to action on the page?
   */
  primary?: boolean;
  /**
   * What background color to use
   */
  backgroundColor?: string;
  /**
   * How large should the button be?
   */
  size?: "small" | "medium" | "large";
  /**
   * Button contents
   */
  label: string;
  /**
   * Optional click handler
   */
  onClick?: () => void;
}

/**
 * Primary UI component for user interaction
 */
export const Button = ({
  primary = false,
  size = "medium",
  backgroundColor,
  label,
  ...props
}: ButtonProps) => {
  const mode = primary
    ? css["storybook-button-primary"]
    : css["storybook-button-secondary"];
  return (
    <button
      type="button"
      className={[
        css["storybook-button"],
        css[`storybook-button-${size}`],
        mode,
      ].join(" ")}
      style={{ backgroundColor }}
      {...props}
    >
      {label}
    </button>
  );
};
