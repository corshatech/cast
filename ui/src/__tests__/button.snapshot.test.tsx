import React from "react";
import renderer from "react-test-renderer";
import { Button } from "../components/Button";

it("renders correctly", () => {
  const tree = renderer.create(<Button label="test" />).toJSON();
  expect(tree).toMatchSnapshot();
});
