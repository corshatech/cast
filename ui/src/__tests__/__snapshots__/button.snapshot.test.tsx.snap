// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="storybook-button storybook-button-medium storybook-button-secondary"
  style={
    {
      "backgroundColor": undefined,
    }
  }
  type="button"
>
  test
</button>
`;
